pyproject.toml

```diff
@@ -117,3 +117,64 @@ Source = "https://github.com/caikit/caikit"
 markers = [
     "examples: marks tests as e2e examples (deselect with '-m \"not examples\"')",
 ]
+
+
+[tool.ruff]
+line-length = 100
+target-version = "py38"
```

Comment:
Interesting. We aim to support 3.8+, but I'm guessing this is the right call so we don't get flagged for a bunch of syntactic improvements in later versions that we can't use, right?

Reply:
Indeed. We previously were targeting 3.9, even if we state that the minimum supported version is 3.8. 

When we drop 3.8, we can bump this up and fix all of the deprecated imports and more, such as 

`from typing import Dict` (we can use `dict` as a type int since 3.9)